odoo/tools/xml_utils.py

```diff
@@ -139,3 +145,133 @@ def leaf_iter(parent_node, node, level):
 
     leaf_iter(None, xml_node, indent_level)
     return xml_node
+
+
+def load_xsd_files_from_url(env, url, xsd_name, force_reload=False,
+                            request_max_timeout=10, xsd_name_prefix='', xsd_names=None, modify_xsd_content=None):
+    """Load XSD file or ZIP archive and save it as ir.attachment.
+
+    If the XSD file/archive has already been saved in database, then just return the attachment.
+    In such a case, the attachment content can also be updated by force if desired.
+    If the attachment is a ZIP archive, then a force reload will also update all attachments from the archive.
+
+    When the attachment is a ZIP archive, every file inside will also be saved as attachments.
+    Filtering which file will be saved can be done by providing a list of `xsd_names`
+
+    The XSD files content can be modified by providing the `modify_xsd_content` function as argument.
+    Typically, this is used when XSD files depend on each other (with the schemaLocation attribute),
+    but it can be used for any purpose.
+
+    :param odoo.api.Environment env: environment of calling module
+    :param str url: URL of XSD file/ZIP archive
+    :param str xsd_name: the name given to the XSD attachment
+    :param bool force_reload: if True, reload the attachment from URL, even if it is already cached
+    :param int request_max_timeout: maximum time (in seconds) before the request times out
+    :param str xsd_name_prefix: if provided, will be added as a prefix to every XSD file name
+    :param list | str xsd_names: if provided, will only save the XSD files with these names
```

Comment:
Naming is not great here, this is easy to confuse with `xsd_name`. Maybe try to find something along the lines of "filter" or "allowlist"?